api/mfa/ceremony_test.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2024 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mfa_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	mfav1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/mfa/v1"
+	"github.com/gravitational/teleport/api/mfa"
+)
+
+func TestPerformMFACeremony(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	testMFAResponse := &proto.MFAAuthenticateResponse{
+		Response: &proto.MFAAuthenticateResponse_TOTP{
+			TOTP: &proto.TOTPResponse{
+				Code: "otp-test-code",
+			},
+		},
+	}
+
+	for _, tt := range []struct {
+		name                   string
+		ceremonyClient         *fakeMFACeremonyClient
+		assertCeremonyResponse func(*testing.T, *proto.MFAAuthenticateResponse, error, ...interface{})
+	}{
+		{
+			name: "OK ceremony success",
+			ceremonyClient: &fakeMFACeremonyClient{
+				challengeResponse: testMFAResponse,
+			},
+			assertCeremonyResponse: func(t *testing.T, mr *proto.MFAAuthenticateResponse, err error, i ...interface{}) {
+				require.NoError(t, err)
+				require.Equal(t, testMFAResponse, mr)
```

Comment:
nit: Use `assert` so it keeps going.

I would also add a message to the assertions, so there is more detail if they fail.

Same for all others.

* https://go.dev/wiki/TestComments#keep-going